source/darwin/secure_transport_tls_channel_handler.c

```diff
@@ -12,23 +12,29 @@
  * express or implied. See the License for the specific language governing
  * permissions and limitations under the License.
  */
-#include <aws/common/encoding.h>
-#include <aws/common/task_scheduler.h>
+#include <aws/io/tls_channel_handler.h>
 
 #include <aws/io/channel.h>
-#include <aws/io/tls_channel_handler.h>
+#include <aws/io/pki_utils.h>
 
+#include <aws/common/encoding.h>
+#include <aws/common/task_scheduler.h>
+
+#include <AvailabilityMacros.h>
 #include <CoreFoundation/CoreFoundation.h>
 #include <Security/SecCertificate.h>
 #include <Security/SecureTransport.h>
 #include <Security/Security.h>
-#include <aws/io/pki_utils.h>
+#include <dlfcn.h>
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wunused-variable"
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
 #pragma clang diagnostic ignored "-Wunused-function"
 
+static OSStatus (*s_SSLSetALPNProtocols_fn_ptr)(SSLContextRef context, CFArrayRef protocols) = NULL;
+static OSStatus (*s_SSLCopyALPNProtocols_fn_ptr)(SSLContextRef context, CFArrayRef *protocols) = NULL;
```

Comment:
[trivial] the _fn_ptr seems redundant, maybe just _fn if you really want?

Reply:
sure